### ISSUE TITLE
[V2] NullReferenceException in PostProcessLayerEditor.OnEnable when selected Prefab

### ISSUE THREAD

**Issue by j4james** [Unity-Technologies/PostProcessing#312]
##### What happened?

1. Created a new project.
2. Added the v2 postprocessing stack to the _Assets_ folder.
3. Added a _Postprocessing Layer_ to the _Main Camera_.
4. Dragged the _Main Camera_ into the _Assets_ folder, converting it into a prefab.
5. With the camera prefab selected, so it's visible in the Inspector, clicked the _Play_ button.
6. At this point I got a `NullReferenceException`.

```
NullReferenceException: Object reference not set to an instance of an object
UnityEditor.Rendering.PostProcessing.PostProcessLayerEditor.OnEnable () (at Assets/PostProcessing/Editor/PostProcessLayerEditor.cs:83)
```

In case I'm out of sync with the latest commit, the line in question is:
`var bundles = m_Target.sortedBundles[evt];`

The null value being the `m_Target.sortedBundles` property, which hasn't been initialised.

Looking at the code, you'd think this would be impossible because just a few lines earlier you've called `m_Target.InitBundles`. But what is happening is that the `PostProcessLayer.haveBundlesBeenInited` field is set to true somehow, even though `m_Bundles` is still null. That looks like it should be impossible, but that's what I'm seeing.

I've been able to work around this by adding an extra check to verify `m_Bundles` is not null in the `PostProcessLayer.InitBundles` and `PostProcessLayer.OnEnable` methods (not sure the latter is strictly necessary), but I'm assuming there should be a better fix if you can figure out why `haveBundlesBeenInited` is incorrectly set in the first place.

##### Post-processing stack version (v1, v2)?
v2

##### Unity version, operating system, target platform (standalone windows, mac, iOS, PS4...)?
Unity 2017.1.0p4 Personal (64bit), Windows 10, no particular target platform (running in the editor).

**Comment by BenjamenJoubert** [comment]
I have experienced the same issue as well and added the same null checks you have. In my case the null reference error happened when a script which manipulates the camera display matrix initiated and executed in Awake or OnEnable. Changing the script to use Start seems to have fixed this issue for now for me.

**Comment by silentlamb1991** [comment]
Oh you reproduced this issue...It's the strangest bug I had seen.

**Comment by Chman** [comment]
The reason for the `haveBundlesBeenInited` variable being used weirdly has to do with a serialization & domain reload issue in Unity. This is unfortunately a known issue I'll have to look at (FYI I won't be able to do much work on post-processing for the next few weeks so don't expect a proper fix until then).

**Comment by Chman** [comment]
Should be fixed in 1c63912. I'll wait for confirmation before closing.

**Comment by j4james** [comment]
@Chman I can't speak for anyone else in this thread, but I can at least confirm that your commit fixes my original test case. Thank you.
